generator/generator-core/src/main/kotlin/io/ia/ignition/module/generator/util/extensions.kt

```diff
@@ -32,7 +37,48 @@ fun String.toPackagePath(maybeScope: ProjectScope? = null): String {
     return if (maybeScope == null) {
         path
     } else {
-        "$path/${maybeScope.name.toLowerCase()}"
+        "$path/${maybeScope.name.lowercase()}"
+    }
+}
+
+fun Path.replacePlaceholders(replacements: Map<String, String> = emptyMap()): Path {
+    this.toFile().replacePlaceholders(replacements)
+    return this
+}
+
+fun File.replacePlaceholders(replacements: Map<String, String> = emptyMap()): String {
+    return this.readText().let { ctnt ->
+        ctnt.lines().map { line ->
+            var updatedLine = line
+
+            replacements.forEach { (key, value) ->
+                if (line.contains(key)) {
+                    updatedLine = line.replace(key, value)
+                }
+            }
+            updatedLine
+        }.joinToString("\n")
+    }.also {
+        this@replacePlaceholders.writeText(it)
+    }
+}
+
+fun InputStream.writeToFileWithReplacements(file: File, replacements: Map<String, String> = emptyMap()) {
+    BufferedReader(InputStreamReader(this)).use { reader ->
+        file.outputStream().use { outStream ->
+            while (reader.ready()) {
+                var line = reader.readLine()
+
+                if (line != null) {
+                    replacements.forEach {
+                        if (line.contains(it.key)) {
+                            line = line.replace(it.key, it.value)
+                        }
+                    }
+                    outStream.write("$line\n".toByteArray(Charsets.UTF_8))
+                }
+            }
+        }
     }
 }
```

Comment:
Should be able to do this file IO cleaner

Reply:
E.G. if you drop the return value:
```kotlin
fun File.replacePlaceholders(replacements: Map<String, String> = emptyMap()) {
    return inputStream().writeToFileWithReplacements(this, replacements)
}

fun InputStream.writeToFileWithReplacements(file: File, replacements: Map<String, String> = emptyMap()) {
    file.bufferedWriter().use { writer ->
        this.bufferedReader().lineSequence().forEach { line ->
            writer.write(
                replacements.entries.fold(line) { currentLine, (key, value) ->
                    currentLine.replace(key, value)
                }
            )
            writer.newLine()
        }
    }
}
```

Reply:
I copy-pasta'd your suggestion for this, and ended up causing a bunch of test failures.  Going to leave the more verbose version in there for now due to lack of time to and will circle back to it in the next round of cleanups.